turbo/transactions/tracing.go

```diff
@@ -70,11 +72,11 @@ func ComputeTxEnv(ctx context.Context, engine consensus.EngineReader, block *typ
 
 	blockContext := core.NewEVMBlockContext(header, core.GetHashFn(header, getHeader), engine, nil, cfg)
 
-	if txIndex == len(block.Transactions()) {
-		// tx is a state sync transaction
-		return nil, blockContext, evmtypes.TxContext{}, statedb, nil, nil
-	}
+	return statedb, blockContext, reader, err
+}
 
+// ComputeTxContext returns the execution environment of a certain transaction.
+func ComputeTxContext(blockContext evmtypes.BlockContext, statedb *state.IntraBlockState, engine consensus.EngineReader, block *types.Block, cfg *chain.Config, txIndex int) (core.Message, evmtypes.TxContext, error) {
```

Comment:
if you don't need `*types.Block` then pls use `*types.Header` - because in Erigon3 many RPC API's are executing individual transactions (instead of whole block) - and for that API's no much reason to read whole `block` from DB.

Reply:
I think for this set of RPCs we read the whole block anyways since we need the header in addition to the transaction. This change would need refactoring the RPC code which is out of the scope of this PR, which is aiming to get state sync transaction tracing working.

Reply:
See it forcing user of API to read block instead of header: https://github.com/erigontech/erigon/pull/12357/files#diff-b13b91974f9eb6b83247e93ff1fa1f29691e84f8191affe0636d77a0cd22ac52R81